torchtext/csrc/register_bindings.cpp

```diff
@@ -129,13 +168,21 @@ TORCH_LIBRARY_FRAGMENT(torchtext, m) {
     .def("PieceToId", &SentencePiece::PieceToId)
     .def("IdToPiece", &SentencePiece::IdToPiece)
     .def_pickle(
+        // The underlying content of SentencePiece contains byte string,
+        // and returing it as std::string cause UTF8 decoding error.
+        // Since TorchScript does not support byte string, we use byte Tensor to
+        // pass around the data.
         // __getstate__
-        [](const c10::intrusive_ptr<SentencePiece> &self) -> std::string {
-          return self->content_;
+        [](const c10::intrusive_ptr<SentencePiece> &self) -> torch::Tensor {
+          auto *data = static_cast<void*>(const_cast<char*>(self->content_.data()));
+          auto numel = static_cast<int64_t>(self->content_.size());
+          return torch::from_blob(data, {numel}, {torch::kUInt8});
```

Comment:
I think it's okay to return a tensor with borrowed memory. The resulting Tensor should be serialized right away.

Reply:
For now, let's return a copy and then remove that copy as a performance optimization later on when we can be 100% sure about this.